File: hammingDistance.cpp
#include <iostream>

int main(int argc, char const *argv[])
{
	// std::cout << "he\n";
	int x = 1, y = 4;

	int xbits[31], ybits[31];
	std::memset(xbits, 0, sizeof xbits);
	std::memset(ybits, 0, sizeof ybits);
	int i = 0;
	int counter = 0;

	while (x > 0)
    {
        xbits[i] = x % 2;
        x = x / 2;
        i++;
    }
	
	i = 0;
    while (y > 0)
    {
        ybits[i] = y % 2;
        y = y / 2;
        i++;
    }

    for(i = 0; i < 31; i++)
    	if (xbits[i] != ybits[i])
    		counter++;

  
    std::cout << counter << std::endl;
   
    return 0;
}

File: hammingDistance.py
def hammingDistance( x, y):
    """
    :type x: int
    :type y: int
    :rtype: int
    """
    binx = bin(x)
    biny = bin(y)
    xbits = [0 for i in range(0,32)]
    ybits = [0 for i in range(0,32)]

    bitsIndex = len(xbits) - 1
    for i in range(  len(binx) - 1, 1, -1):
        xbits[bitsIndex] = binx[i]
        bitsIndex -= 1
    
    bitsIndex = len(ybits) - 1
    for i in range( len(biny) - 1, 1, -1):
        ybits[bitsIndex] = biny[i]
        bitsIndex -= 1

    counter = 0
    for i in range(0 , len(xbits), 1):
        if int(xbits[i]) != int(ybits[i]):
            counter += 1

    print(counter)
if __name__ == '__main__':
    hammingDistance(1,4)
